lib/bitstyles_phoenix/components/icon.ex

```diff
@@ -11,26 +11,48 @@ defmodule BitstylesPhoenix.Components.Icon do
 
   `opts[:name]` — The name of the icon to render. This must be present in the SVG that renders all the icons in the top of the body e.g. `_svgs.html.eex` as a `<symbol>` containing one filled path.
 
+  `opts[:size]` - Specify the icon size to use. Available sizes are specified in CSS, and default to `s`, `m`, `l`, `xl`. If you do not specify a size, the icon will fit into a `1em` square.
+
   ## Examples
 
       iex> safe_to_string ui_icon("right")
-      ~s(<svg alt="" class="a-icon" role="presentation"><use xlink:href="#icon-right"></svg>)
+      ~s(<svg aria-hidden="true" class="a-icon" focusable="false" height="16" width="16" xmlns="http://www.w3.org/2000/svg"><use xlink:href="#icon-right"></svg>)
 
       iex> safe_to_string ui_icon("right", size: "s")
-      ~s(<svg alt="" class="a-icon a-icon--s" role="presentation"><use xlink:href="#icon-right"></svg>)
+      ~s(<svg aria-hidden="true" class="a-icon a-icon--s" focusable="false" height="16" width="16" xmlns="http://www.w3.org/2000/svg"><use xlink:href="#icon-right"></svg>)
 
       iex> safe_to_string ui_icon("trashcan")
-      ~s(<svg alt="" class="a-icon" role="presentation"><use xlink:href="#icon-trashcan"></svg>)
+      ~s(<svg aria-hidden="true" class="a-icon" focusable="false" height="16" width="16" xmlns="http://www.w3.org/2000/svg"><use xlink:href="#icon-trashcan"></svg>)
+
+      iex> safe_to_string ui_icon("trashcan", width: 20)
+      ~s(<svg aria-hidden="true" class="a-icon" focusable="false" height="20" width="20" xmlns="http://www.w3.org/2000/svg"><use xlink:href="#icon-trashcan"></svg>)
 
       iex> safe_to_string ui_icon("trashcan", class: "foo bar")
-      ~s(<svg alt="" class="a-icon foo bar" role="presentation"><use xlink:href="#icon-trashcan"></svg>)
+      ~s(<svg aria-hidden="true" class="a-icon foo bar" focusable="false" height="16" width="16" xmlns="http://www.w3.org/2000/svg"><use xlink:href="#icon-trashcan"></svg>)
   """
   def ui_icon(name, opts \\ []) do
     classname =
       classnames(["a-icon", {"a-icon--#{opts[:size]}", opts[:size] != nil}, opts[:class]])
 
-    content_tag(:svg, class: classname, alt: "", role: "presentation") do
+    opts = opts |> put_default_width()
+
+    content_tag(:svg,
+      class: classname,
+      "aria-hidden": "true",
+      focusable: "false",
+      xmlns: "http://www.w3.org/2000/svg",
+      width: opts[:width],
+      height: opts[:width]
+    ) do
       tag(:use, "xlink:href": "#icon-#{name}")
     end
   end
+
+  defp put_default_width(opts) do
+    if is_integer(opts[:width]) do
+      opts
+    else
+      opts |> Keyword.put_new(:width, 16)
+    end
```

Comment:
I think the default values are still a bit odd here. I'd rather pass the opts all to the outer element, overwriting what we want to set always and leaving alone what needs defaults with `put_new`. I am not quite sure what the `is_integer` check is for :) I think if we pass in a value we should take that value rather than the default, no matter if it is an integer or not.

```suggestion
    opts = opts
        |> put_defaults()
        |> Keyword.put(:class, classname)
        |> Keyword.merge("aria-hidden": "true",
                                         focusable: "false",
                                          xmlns: "http://www.w3.org/2000/svg")

    content_tag(:svg, opts) do
      tag(:use, "xlink:href": "#icon-#{name}")
    end
  end

 @default_size 16
  defp put_defaults(opts) do
      opts 
      |> Keyword.put_new(:width, @default_size)
      |> Keyword.put_new(:height, @default_size)
   end
```

Reply:
Thanks! That works well 🎉